spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/leader/LeaderInitiator.java

```diff
@@ -263,30 +273,37 @@ protected class LeaderSelector implements Callable<Void> {
 
 		protected final String role = LeaderInitiator.this.candidate.getRole();
 
-		private volatile boolean locked = false;
+		private volatile boolean leader = false;
 
 		@Override
 		public Void call() {
 			try {
 				while (isRunning()) {
 					try {
-						if (LeaderInitiator.this.lock.isLocked()) {
+						if (logger.isTraceEnabled()) {
+							logger.trace("Am I the leader (" + LeaderInitiator.this.candidate.getRole() + ") ? "
+									+ this.leader);
+						}
+						if (LeaderInitiator.this.getLock().isLockedByCurrentThread()) {
+							if (!this.leader) {
+								// Since we have the lock we need to ensure that the leader flag is set
+								this.leader = true;
+							}
 							// Give it a chance to expire.
 							if (yieldSign.tryAcquire(LeaderInitiator.this.heartBeatMillis, TimeUnit.MILLISECONDS)) {
 								revokeLeadership();
 								// Give it a chance to elect some other leader.
 								Thread.sleep(LeaderInitiator.this.busyWaitMillis);
 							}
-						}
+						} 
 						else {
 							// We try to acquire the lock
-							boolean acquired =
-									LeaderInitiator.this.lock.tryLock(LeaderInitiator.this.heartBeatMillis,
-											TimeUnit.MILLISECONDS);
-							if (!this.locked) {
-								if (acquired) {
+							boolean acquired = LeaderInitiator.this.getLock()
+									.tryLock(LeaderInitiator.this.heartBeatMillis, TimeUnit.MILLISECONDS);
+							if (acquired) {
+								if (!this.leader) {
```

Comment:
Looks like these to `ifs` can be combined to a single one.
Do I  miss anything?

Reply:
Yes it can, however It was to `ifs` in the original code as well. But sure it can be one if-statement.